darasimiolaifa/traveller-fin-compass
# feat(Conversion): implement currency conversion feature

## What does this PR do?
This PR implements the conversion functionality of the app

## What major activities were carried out?
- create actions to handle form events
- create types to represent form events
- create reducers to handle form actions
- create redux store to handle all state

## How can this be manually tested?
- Clone the repository
- Fetch the branch
- Install the dependencies
- cd into the `traveller-fin-compass` folder
- Start project by running `npm start` in the terminal
- Your browser should automatically load to show you the homepage screen below:

![final homepage](https://user-images.githubusercontent.com/28527393/72210159-6bc5b500-34b7-11ea-893f-2498440b0e1a.png)
- Wait for the currencies to load. Then pick the base and foreign currencies, enter the amount to convert, and click the convert button. After  while you should see your converted amount, just like in the picture below:

![Conversion](https://user-images.githubusercontent.com/28527393/72209358-87c45900-34ad-11ea-8d23-b44d13bc5780.png)
- Click the `Invert conversion` button and it should invert the conversion
- Click the `Reset conversion` button, and it should clear the amount input and set the conversion amount back to 0

PT Story Link: [#168503783](https://www.pivotaltracker.com/story/show/168503783)
